Created README_hd.md

AkshitGulyan opened the pull request:
A Hindi Translation for README

# What does this PR do?
It adds the Hindi Translation for the README File !
<!--
Congratulations! You've made it this far! You're not quite done yet though.

Once merged, your PR is going to appear in the release notes with the title you set, so make sure it's a great title that fully reflects the extent of your awesome contribution.

Then, please replace this with a description of the change and which issue is fixed (if applicable). Please also include relevant motivation and context. List any dependencies (if any) that are required for this change.

Once you're done, someone will review your PR shortly (see the section "Who can review?" below to tag some potential reviewers). They may suggest changes to make the code even better. If no one reviewed your PR after a week has passed, don't hesitate to post a new comment @-mentioning the same persons---sometimes notifications get lost.
-->

<!-- Remove if not applicable -->

Fixes # (issue)


## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?


## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

<!-- Your PR will be replied to more quickly if you can figure out the right person to tag with @

 If you know how to use git blame, that is the easiest way, otherwise, here is a rough guide of **who to tag**.
 Please tag fewer than 3 people.

Models:

- albert, bert, xlm: @LysandreJik
- blenderbot, bart, marian, pegasus, encoderdecoder,  t5: @patrickvonplaten, @patil-suraj
- longformer, reformer, transfoxl, xlnet: @patrickvonplaten
- fsmt: @stas00
- funnel: @sgugger
- gpt2: @patrickvonplaten, @LysandreJik
- rag: @patrickvonplaten, @lhoestq
- tensorflow: @LysandreJik

Library:

- benchmarks: @patrickvonplaten
- deepspeed: @stas00
- ray/raytune: @richardliaw, @amogkam
- text generation: @patrickvonplaten
- tokenizers: @n1t0, @LysandreJik
- trainer: @sgugger
- pipelines: @LysandreJik

Documentation: @sgugger

HF projects:

- datasets: [different repo](https://github.com/huggingface/datasets)
- rust tokenizers: [different repo](https://github.com/huggingface/tokenizers)

Examples:

- maintained examples (not research project or legacy): @sgugger, @patil-suraj
- research_projects/bert-loses-patience: @JetRunner
- research_projects/distillation: @VictorSanh

 -->

AkshitGulyan committed:
Created README_hd.md

A Hindi Translation for README

sgugger reviewed (commented):
Can you also add the proper info in [this dictionary](https://github.com/huggingface/transformers/blob/7a1c68a8454c25c55f3f8978c182ea90e3412f5c/utils/check_copies.py#L39) so this translation is kept up to date when we add new models?

AkshitGulyan commented:
added the proper info in [this dictionary](https://github.com/huggingface/transformers/blob/7a1c68a8454c25c55f3f8978c182ea90e3412f5c/utils/check_copies.py#L39)

By a new pull request 
[Update check_copies.py #19905](https://github.com/huggingface/transformers/pull/19905)

sgugger commented:
No this should all be in the same pull request please.

AkshitGulyan committed:
updated check_copies.py

Added the Proper info for Hindi Translation of README File !

AkshitGulyan commented:
> No this should all be in the same pull request please.

Updated the check_copies.py in this current Pull Request and closed the previous Pull Request !

pacman100 commented on README_hd.md:
```diff
@@ -0,0 +1,417 @@
+
+सबसे पहले, पायथन के उस संस्करण के साथ एक आभासी वातावरण बनाएं जिसका आप उपयोग करने और उसे सक्रिय करने की योजना बना रहे हैं।
+
+फिर, आपको Flax, PyTorch या TensorFlow में से किसी एक को स्थापित करने की आवश्यकता है। अपने प्लेटफ़ॉर्म पर इन फ़्रेमवर्क को स्थापित करने के लिए, [TensorFlow स्थापना पृष्ठ](https://www.tensorflow.org/install/), [PyTorch स्थापना पृष्ठ](https://pytorch.org/get-started /locally/# देखें) start-locally) या [Flax install page](https://github.com/google/flax#quick-install).
```
`install page` is inconsitent across framework links

pacman100 replied:
I meant Tensorflow and PyTorch has "स्थापना पृष्ठ" but Flax has "install page"

AkshitGulyan replied:
Done !

pacman100 commented on README_hd.md:
```diff
@@ -0,0 +1,417 @@
+जब इनमें से कोई एक बैकएंड सफलतापूर्वक स्थापित हो जाता है, तो ट्रांसफॉर्मर निम्नानुसार स्थापित किए जा सकते हैं:
+
+`` `बाश
+पाइप स्थापित ट्रांसफार्मर
```
original command in English?

pacman100 replied:
the cmd should be in english so that user can copy it and run?

AkshitGulyan replied:
Updated !

pacman100 commented on README_hd.md:
````diff
@@ -0,0 +1,417 @@
+ट्रांसफॉर्मर कोंडा के माध्यम से निम्नानुसार स्थापित किया जा सकता है:
+
+``` खोल स्क्रिप्ट
+कोंडा इंस्टाल-सी हगिंगफेस ट्रांसफॉर्मर
````
command in English?

pacman100 replied:
the cmd should be in english so that user can copy it and run?

AkshitGulyan replied:
Updated !

pacman100 commented on README_hd.md:
```diff
@@ -0,0 +1,417 @@
+
+कोड की दूसरी पंक्ति पाइपलाइन द्वारा उपयोग किए गए पूर्व-प्रशिक्षित मॉडल को डाउनलोड और कैश करती है, जबकि कोड की तीसरी पंक्ति दिए गए पाठ पर मूल्यांकन करती है। यहां उत्तर 99 आत्मविश्वास के स्तर के साथ "सकारात्मक" है।
+
+कई एनएलपी कार्यों में बॉक्स से बाहर पाइपलाइनों का पूर्व-प्रशिक्षण होता है। उदाहरण के लिए, हम किसी दिए गए पाठ से किसी प्रश्न का उत्तर आसानी से निकाल सकते हैं:
```
"out of the box" is translated in the literal sense which isn't quite understandable. Also, this deviates from the original readmedoc having vision example here

AkshitGulyan replied:
Fixed !
I wrote the out of the box in devanagari script,
Now i don't think anyone will find it confusing.

pacman100 commented on README_hd.md:
```diff
@@ -0,0 +1,417 @@
+| [दस्तावेज़ीकरण](https://huggingface.co/transformers/) | पूरा एपीआई दस्तावेज़ीकरण और ट्यूटोरियल |
+| [कार्य सारांश](https://huggingface.co/docs/transformers/task_summary) | ट्रांसफॉर्मर समर्थित कार्य |
+| [प्रीप्रोसेसिंग ट्यूटोरियल](https://huggingface.co/docs/transformers/preprocessing) | मॉडल के लिए डेटा तैयार करने के लिए `टोकनाइज़र` का उपयोग करना |
+| [प्रशिक्षण और फाइन-ट्यूनिंग](https://huggingface.co/docs/transformers/training) | PyTorch/TensorFlow के ट्रेनिंग लूप या `ट्रेनर` API में ट्रांसफॉर्मर द्वारा दिए गए मॉडल का उपयोग करें |
```
here fine-tuning is written in Devanagari script which should be followed in other places above in the main content where it was translated as fixing

AkshitGulyan replied:
Fixed !

pacman100 commented on README_hd.md:
```diff
@@ -0,0 +1,417 @@
+| [प्रीप्रोसेसिंग ट्यूटोरियल](https://huggingface.co/docs/transformers/preprocessing) | मॉडल के लिए डेटा तैयार करने के लिए `टोकनाइज़र` का उपयोग करना |
+| [प्रशिक्षण और फाइन-ट्यूनिंग](https://huggingface.co/docs/transformers/training) | PyTorch/TensorFlow के ट्रेनिंग लूप या `ट्रेनर` API में ट्रांसफॉर्मर द्वारा दिए गए मॉडल का उपयोग करें |
+| [क्विक स्टार्ट: ट्वीकिंग एंड यूज़ केस स्क्रिप्ट्स](https://github.com/huggingface/transformers/tree/main/examples) | विभिन्न कार्यों के लिए केस स्क्रिप्ट का उपयोग करें |
+| [मॉडल साझा करना और अपलोड करना](https://huggingface.co/docs/transformers/model_sharing) | समुदाय के साथ अपने ठीक-ठाक मॉडल अपलोड और साझा करें |
```
okay-okay as translation for fine-tuned

AkshitGulyan replied:
Fixed !

pacman100 reviewed (commented):
Hello @AkshitGulyan, great work 🤗. Thanks for putting in a lot of effort for making  `transformers` accessible to an even larger number of people who are comfortable in Hindi language ✨. Looking good, left comments for improving readability.

AkshitGulyan commented on README_hd.md:
```diff
@@ -0,0 +1,417 @@
+
+🤗 Transformers त्वरित डाउनलोड और उपयोग के लिए एक एपीआई प्रदान करता है, जिससे आप किसी दिए गए पाठ पर एक पूर्व-प्रशिक्षित मॉडल ले सकते हैं, इसे अपने डेटासेट पर ठीक कर सकते हैं और इसे [मॉडल हब] (https://huggingface.co/models) के माध्यम से समुदाय के साथ साझा कर सकते हैं। ) . इसी समय, प्रत्येक परिभाषित पायथन मॉड्यूल पूरी तरह से स्वतंत्र है, जो संशोधन और तेजी से अनुसंधान प्रयोगों के लिए सुविधाजनक है।
+
+🤗 Transformers तीन सबसे लोकप्रिय गहन शिक्षण पुस्तकालयों का समर्थन करता है： [Jax](https://jax.readthedocs.io/en/latest/), [PyTorch](https://pytorch.org/) and [TensorFlow](https://www.tensorflow.org/) — और इसके साथ निर्बाध रूप से एकीकृत होता है। आप अपने मॉडल को सीधे एक ढांचे के साथ प्रशिक्षित कर सकते हैं और दूसरे के साथ लोड और अनुमान लगा सकते हैं।
```
yes it's a little confusing but i thought that i should translate each and every bit of the file !

AkshitGulyan replied:
Fixed !

AkshitGulyan commented on README_hd.md:
```diff
@@ -0,0 +1,417 @@
+- [डिस्टिलबर्ट के साथ प्रश्नोत्तर](https://huggingface.co/distilbert-base-uncased-distilled-squad?text=Which+name+is+also+used+to+describe+the+Amazon+rainforest+in+English%3F&context=The+Amazon+rainforest+%28Portuguese%3A+Floresta+Amaz%C3%B4nica+or+Amaz%C3%B4nia%3B+Spanish%3A+Selva+Amaz%C3%B3nica%2C+Amazon%C3%ADa+or+usually+Amazonia%3B+French%3A+For%C3%AAt+amazonienne%3B+Dutch%3A+Amazoneregenwoud ...
+- [अनुवाद के लिए T5 का प्रयोग करें](https://huggingface.co/t5-base?text=My+name+is+Wolfgang+and+I+live+in+Berlin)
+
+**[Write With Transformer](https://transformer.huggingface.co)**，हग फेस टीम द्वारा बनाया गया, यह एक आधिकारिक पाठ पीढ़ी है demo。
```
oops

AkshitGulyan replied:
Fixed !

AkshitGulyan commented on README_hd.md:
```diff
@@ -0,0 +1,417 @@
+
+टोकननाइज़र सभी पूर्व-प्रशिक्षित मॉडलों के लिए प्रीप्रोसेसिंग प्रदान करता है और इसे सीधे एक स्ट्रिंग (जैसे ऊपर दिए गए उदाहरण) या किसी सूची पर बुलाया जा सकता है। यह एक डिक्शनरी (तानाशाही) को आउटपुट करता है जिसे आप डाउनस्ट्रीम कोड में उपयोग कर सकते हैं या `**` अनपैकिंग एक्सप्रेशन के माध्यम से सीधे मॉडल को पास कर सकते हैं।
+
+मॉडल स्वयं एक नियमित [Pytorch `nn.Module`](https://pytorch.org/docs/stable/nn.html#torch.nn.Module) या [TensorFlow `tf.keras.Model`](https ://pytorch.org/docs/stable/nn.html#torch.nn.Module) ://www.tensorflow.org/api_docs/python/tf/keras/Model) (आपके बैकएंड के आधार पर), जो हो सकता है सामान्य तरीके से उपयोग किया जाता है। [यह ट्यूटोरियल](https://huggingface.co/transformers/training.html) बताता है क ...
```
@pacman100 
I really appreciate your contribution
Can you please edit the things which you are feeling that they should be formed in a different way !

AkshitGulyan replied:
Fixed !

AkshitGulyan committed:
updated README_hd.md

Fixed some translation issues !

AkshitGulyan commented on README_hd.md:
```diff
@@ -0,0 +1,417 @@
+</h4>
+
+<h3 align="center">
+    <p>Jax, PyTorch और TensorFlow के लिए उन्नत प्राकृतिक भाषा संसाधन</p>
```
Fixed !

AkshitGulyan commented on README_hd.md:
```diff
@@ -0,0 +1,417 @@
+
+**[Write With Transformer](https://transformer.huggingface.co)**，हग फेस टीम द्वारा बनाया गया, यह एक आधिकारिक पाठ पीढ़ी है demo。
+
+## यदि आप हगफेस टीम से बीस्पोक समर्थन की तलाश कर रहे हैं
```
Fixed !

AkshitGulyan commented on README_hd.md:
```diff
@@ -0,0 +1,417 @@
+
+1.मॉडल जीवनचक्र के हर हिस्से को शामिल करता है:
+    - कोड की केवल 3 पंक्तियों में उन्नत मॉडलों को प्रशिक्षित करें
+    - मॉडल को मनमाने ढंग से विभिन्न गहन शिक्षण ढांचे के बीच स्थानांतरित किया जा सकता है, जैसा आप चाहते हैं
```
Fixed !

AkshitGulyan commented on README_hd.md:
```diff
@@ -0,0 +1,417 @@
+1. उपयोग में आसानी के लिए उन्नत मॉडल:
+    - एनएलयू और एनएलजी पर बेहतर प्रदर्शन
+    - प्रवेश के लिए कम बाधाओं के साथ शिक्षण और अभ्यास के अनुकूल
+    - उच्च स्तरीय अमूर्तता, केवल तीन वर्गों को जानने की जरूरत है
```
Fixed !

AkshitGulyan commented on README_hd.md:
```diff
@@ -0,0 +1,417 @@
+
+## मुझे ट्रांसफॉर्मर का उपयोग कब नहीं करना चाहिए?
+
+- यह लाइब्रेरी मॉड्यूलर न्यूरल नेटवर्क टूलबॉक्स नहीं है। मॉडल फ़ाइल में कोड जानबूझकर अल्पविकसित है, बिना अतिरिक्त सार इनकैप्सुलेशन के, ताकि शोधकर्ता अमूर्तता और फ़ाइल जंपिंग में शामिल हुए बिना जादुई परिवर्तनों के माध्यम से जल्दी से पुनरावृति कर सकें।
```
Fixed !

AkshitGulyan commented on README_hd.md:
```diff
@@ -0,0 +1,417 @@
+
+## स्थापित करना
+
+### पाइप का उपयोग करना
```
Fixed !

AkshitGulyan commented:
Any Update ?

sgugger commented:
I think you need to run `make fix-copies` on your side to adjust the READMEs, then it should be good to merge if all comments are addressed :-)

pacman100 commented:
Please address remaining comments and then we are good to go

AkshitGulyan committed:
Update README_hd.md

AkshitGulyan committed:
Update README_hd.md

AkshitGulyan committed:
Update README_hd.md

AkshitGulyan commented:
> Please address remaining comments along with steps Sylvain has mentioned and then we are good to go

Addressed all the comments and updated the file according to them !
Can you please help me understanding this fix-copies concept which Sylvain has mentioned as i dont know about it !

pacman100 commented:
Hello @AkshitGulyan, in the above PR I fixed subtle and time-consuming bugs to run `make fix-copies` without any issues. The details are below so that you can do these things next time.

1. When I ran `make fix-copies` locally I got below error:
```
(ml) sourabmangrulkar@Sourabs-MacBook-Pro transformers % make fix-copies
python utils/check_copies.py --fix_and_overwrite
Traceback (most recent call last):
  File "/Users/sourabmangrulkar/Code/transformers/utils/check_copies.py", line 572, in <module>
    check_copies(args.fix_and_overwrite)
  File "/Users/sourabmangrulkar/Code/transformers/utils/check_copies.py", line 270, in check_copies
    check_model_list_copy(overwrite=overwrite)
  File "/Users/sourabmangrulkar/Code/transformers/utils/check_copies.py", line 455, in check_model_list_copy
    localized_md_list = get_model_list(filename, _start_prompt, _end_prompt)
  File "/Users/sourabmangrulkar/Code/transformers/utils/check_copies.py", line 303, in get_model_list
    while not lines[start_index].startswith(start_prompt):
IndexError: list index out of range
make: *** [fix-copies] Error 1
```  

2. After spending time diving into `utils/check_copies.py` found the issue wherein `prompt_start` specified was not matching to the line in `README_hd.md`. Made them same.

3. Then got this issue:
```
Traceback (most recent call last):
  File "/Users/sourabmangrulkar/Code/transformers/utils/check_copies.py", line 354, in convert_to_localized_md
    localized_model_index = {
  File "/Users/sourabmangrulkar/Code/transformers/utils/check_copies.py", line 355, in <dictcomp>
    re.search(r"\*\*\[([^\]]*)", line).groups()[0]: line
AttributeError: 'NoneType' object has no attribute 'groups'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/sourabmangrulkar/Code/transformers/utils/check_copies.py", line 575, in <module>
    check_copies(args.fix_and_overwrite)
  File "/Users/sourabmangrulkar/Code/transformers/utils/check_copies.py", line 270, in check_copies
    check_model_list_copy(overwrite=overwrite)
  File "/Users/sourabmangrulkar/Code/transformers/utils/check_copies.py", line 459, in check_model_list_copy
    readmes_match, converted_md_list = convert_to_localized_md(md_list, localized_md_list, _format_model_list)
  File "/Users/sourabmangrulkar/Code/transformers/utils/check_copies.py", line 359, in convert_to_localized_md
    raise AttributeError("A model name in localized READMEs cannot be recognized.")
AttributeError: A model name in localized READMEs cannot be recognized.
(ml) sourabmangrulkar@Sourabs-MacBook-Pro transformers % python utils/check_copies.py
Traceback (most recent call last):
  File "/Users/sourabmangrulkar/Code/transformers/utils/check_copies.py", line 351, in convert_to_localized_md
    localized_model_index = {
  File "/Users/sourabmangrulkar/Code/transformers/utils/check_copies.py", line 352, in <dictcomp>
    re.search(r"\*\*\[([^\]]*)", line).groups()[0]: line
AttributeError: 'NoneType' object has no attribute 'groups'
```

This was a subtle bug which took quite some time to figure out. You had improperly formatted the following models with improper spaces resulting in regex failing, below shows the buggy version:
```
1. ** [TrOCR] (https://huggingface.co/docs/transformers/model_doc/trocr) ** (from Microsoft) released with the paper [TrOCR: Transformer-based Optical Character Recognition with Pre-trained Models](https://arxiv.org/abs/2109.10282) by Minghao Li, Tengchao Lv, Lei Cui, Yijuan Lu, Dinei Florencio, Cha Zhang, Zhoujun Li, Furu Wei.
1. ** [UL2] (https://huggingface.co/docs/transformers/model_doc/ul2) ** (from Google Research) released with the paper [Unifying Language Learning Paradigms](https://arxiv.org/abs/2205.05131v1) by Yi Tay, Mostafa Dehghani, Vinh Q. Tran, Xavier Garcia, Dara Bahri, Tal Schuster, Huaixiu Steven Zheng, Neil Houlsby, Donald Metzler 
```

So, after fixing it everything works as expected:
```
(ml) sourabmangrulkar@Sourabs-MacBook-Pro transformers % make fix-copies             
python utils/check_copies.py --fix_and_overwrite
python utils/check_table.py --fix_and_overwrite
python utils/check_dummies.py --fix_and_overwrite
```

Also, many model list is very very inconsistent with some having names in Hindi while others in English. Follow the format where all model names are in latin script instead of Devanagari script.

pacman100 commented:
Hello @AkshitGulyan, please transfer the changes from above sample PR to this PR. Thank you and hope the above explanation clarifies the steps that Sylvain was suggesting.

AkshitGulyan committed:
Update check_copies.py
